### ISSUE TITLE
logger instead of logging

### ISSUE THREAD

**Issue by markus-hinsche** [Welthungerhilfe/cgm-ml#330]
**Problem:** It is not easily configurable to see the logs in `src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/src/train.py`. This can be verified by running this file and observing the logs to stdout.
Using `logging.basicConfig()` seems to cause problems when other modules are also defining loggers, handlers, etc.

I suggest the following change:

Instead
```
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s - %(pathname)s: line %(lineno)d')
logging.info('test log line')
```
we could write:
```
logger = logging.getLogger(__name__)
logger.setLevel(logging.INFO)
logger.info('test log line')
```

**Comment by jasminziegler** [comment]
@markus-hinsche thanks for the hint! Is the goal to only see logs from this file? and not from other modules? I see some logs from train.py and from tensorflow for example.

**Comment by markus-hinsche** [comment]
The goal is to see the logs from this module (e.g. train.py) in addition to other (e.g. tensorflow) logs

**Comment by jasminziegler** [comment]
![grafik](https://user-images.githubusercontent.com/29899146/116518938-2c4ff180-a8d1-11eb-9b1c-7f5b67914ccf.png)
I can actually see them all. But @prajurock has a similar issue with evaluation logs!

**Comment by jasminziegler** [comment]
Setup my system with Python 3.79, still cannot reproduce the issue 
![grafik](https://user-images.githubusercontent.com/29899146/124116122-e220f300-da6e-11eb-9416-990d4c820d4e.png)

**Comment by jasminziegler** [comment]
updated logging in #403 and this is what I observed:
- cannot see anymore logs while running train.py locally
- cannot see any logs while running training on cluster

**Comment by jasminziegler** [comment]
added 
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s - %(pathname)s: line %(lineno)d')
back in, works locally. testing on cluster now.

**Comment by markus-hinsche** [comment]
How do other (e.g. open-source) projects solve this?

**Comment by jasminziegler** [comment]
- https://github.com/scikit-learn/scikit-learn/search?q=logging
-

**Comment by jasminziegler** [comment]
Not setting the root logger to INFO will only show warning or error msg (or above) since the default logging level is warning.

**Comment by jasminziegler** [comment]
this does not work 
![grafik](https://user-images.githubusercontent.com/29899146/125759475-cb161452-fe88-47d0-a481-95f05ccf3100.png)

this does not work
![grafik](https://user-images.githubusercontent.com/29899146/125759634-c83fb6b5-571e-4f6b-8f6a-b4c1cd6a5f5f.png)

and this works
![grafik](https://user-images.githubusercontent.com/29899146/125759875-6799f89c-af30-494a-9a82-7a63e2da03f0.png)

**Comment by jasminziegler** [comment]
Found a way to get around the usage of "basicConfig": 
![grafik](https://user-images.githubusercontent.com/29899146/125761660-10208693-d5be-4b10-80f5-aeb5ef9ec6d2.png)

What do you think about this solution to define handlers and formatters every time, @markus-hinsche ? By doing so, we can prevent using basicConfig.
Going through other opensource projects was not helpful if we do not want to set up entire logging modules. Not sure if that is worth the effort, but happy to hear your opinion about this

**Comment by markus-hinsche** [comment]
Looks like a good solution to define handler, formatter, logger etc.

You can also use a config file to avoid clutter in the python configs. 
https://stackoverflow.com/questions/4441842/python-logging-configuration-file

**Comment by jasminziegler** [comment]
that's funny, we end up with the very first solution again :-) but learned a lot on the way. 

Can you put into some words how and why you think that basicConfig causes/might cause issues? 
Good to remember for the future :)

**Comment by jasminziegler** [comment]
this was closed in #403
